scripts/ui.js

```diff
@@ -317,7 +317,7 @@ function clearProgress() {
  */
 function validateUrl(url) {
     const validurl = new RegExp('^(https?:\\/\\/)?' + // protocol
-        '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+        '((([a-z\\d]|[a-z\\d][a-z\\d-]*[a-z\\d])\\.)*([a-z\\d]|[a-z\\d][a-z\\d-]*[a-z\\d])|' + // domain name
```

Comment:
```suggestion
        '(([a-z\\d]([a-z\\d-]*[a-z\\d])?\\.)*([a-z\\d]([a-z\\d-]*[a-z\\d])?)|' + // domain name
```
After (finally) taking a look at this I'd like to propose this slightly shorter alternative which I believe should be functionally identical to your proposed change. Could you do a quick sanity check for me if that's actually the case?

Otherwise it seems good and the change makes sense to me

Reply:
I can't see any issues from my end, but I imagine top-cat has more regex experience than me